### ISSUE TITLE
Crash when revoking certificate

### ISSUE THREAD

**Issue by nvx** [remilapeyre/vault-acme#7]
Creating a certificate, then copying the lease id returned and running:

```
vault lease revoke acme/certs/test/...
```

Results in a panic in the plugin. The offending line appears to be https://github.com/remilapeyre/vault-acme/blob/6ac7a952d9f7d04354aa4ecc0cc2be963e7da411/acme/secret_cert.go#L56 with the reason:

```
panic: interface conversion: interface {} is string, not []uint8
```

Interestingly the next line then casts it to a string anyway.

**Comment by remilapeyre** [comment]
Thanks @nvx ! This must be where all the errors I had must be coming. What's weird is that revoking certificates is tested in `checkRevokeCert()`.

I will try to have a look at this this week.

**Comment by remilapeyre** [comment]
Hi @nvx, this has been fixed in 7a94b6d44355dc2a2dae8c5bd86ef742fd3af88d.

I think you will need to force-revoke the current lease before upgrading the plugin: `vault write -force /sys/leases/revoke-force/acme`.
